### DISCUSSION TITLE
What specification of machine do Microsoft Employees use when working on Roslyn?

### DISCUSSION THREAD

**Discussion by 06needhamt** [dotnet/roslyn#47035] [General] [answered]
Just out of curiosity I was wondering what spec of machine Microsoft employees use when working on Roslyn or what tips you have of making visual studio perform better while working on Roslyn?

My machine Specs:
![image](https://user-images.githubusercontent.com/4162227/90907881-82c47200-e3cb-11ea-942c-68f598085831.png)

With this machine, it takes ~5 minutes to load up the Roslyn.sln file in visual studio and about 5 more minutes before it becomes usable. Even after this, there is a noticeable delay in keystrokes when typing in a file.

I am interested in making more contributions but I am put off due to the slowness of Visual Studio.

**Comment by CyrusNajmabadi** [comment]
@06needhamt can you report a trace here https://github.com/dotnet/roslyn/wiki/Recording-performance-traces-with-PerfView

I'm on a dual core laptop and I'm not experiencing the issues your taking about. Note: I do generally give vs about 1+ minute to load and get some with initial work. But from that point on, things seem to work ok.

**Comment by Youssef1313** [comment]
I think it would be super awesome if there is a way to work on Roslyn from VS Code, it's much much lighter.

**Comment by jmarolf** [comment]
most of the compiler team uses VS Code nowadays. As Cyrus says, without a perf trace its hard to guess at what is happening.

**Comment by Youssef1313** [comment]
@jmarolf, Are there any instructions/docs on using VS Code with Roslyn solution?

**Comment by jmarolf** [comment]
@333fred or @RikkiGibson could either of you update https://github.com/dotnet/roslyn/tree/master/docs/contributing with the details about your Vs Code workflows?

**Reply by 333fred** [reply]
I'll try and do something on Monday, but the process is very simple. Clone, run restore, open in vscode.

**Reply by jmarolf** [reply]
perhaps running tests might not be obvious to new contributors?

**Reply by RikkiGibson** [reply]
Instructions have been added at https://github.com/dotnet/roslyn/blob/master/docs/contributing/Building,%20Debugging,%20and%20Testing%20on%20Unix.md#working-in-visual-studio-code

/cc @jmarolf @Youssef1313

**Comment by sharwell** [comment, marked as answer]
The system specifications vary widely from laptops to desktops. I would say the keys to a good experience are:

* SSD minimum, NVMe preferred (both OS/install drive and the location where Roslyn is checked out).
* 16G memory minimum, 32G preferred.
* 4 cores + hyperthreading minimum, 8+ cores preferred.

**Reply by Youssef1313** [reply]
>     * SSD minimum, NVMe preferred (both OS/install drive and the location where Roslyn is checked out).

+1 For SSD. I worked on an Azure VM that's nearly better than my laptop by the SSD only. There is a **super super super great** improvement.
